packages/store/src/utils.js

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+
+const { getPrototypeOf, create, fromEntries } = Object;
+const { ownKeys, apply } = Reflect;
+
+const compareStringified = (left, right) => {
+  left = String(left);
+  right = String(right);
+  // eslint-disable-next-line no-nested-ternary
+  return left < right ? -1 : left > right ? 1 : 0;
+};
+
+/**
+ * @param {object} obj
+ * @returns {(string|symbol)[]}
+ */
+export const getMethodNames = obj => {
+  const result = [];
+  while (obj !== null && obj !== Object.prototype) {
+    const mNames = ownKeys(obj).filter(name => typeof obj[name] === 'function');
+    result.push(...mNames);
+    obj = getPrototypeOf(obj);
+  }
+  result.sort(compareStringified);
+  return harden(result);
+};
+harden(getMethodNames);
+
+export const bindAllMethods = obj =>
```

Comment:
I don't understand what bindAllMethods is achieving just from reading the code. Please add a comment.  

Why is it needed in stakeFactoryKit?  Why is it only needed there?

Reply:
Done. You may want to have a look.

Reply:
Thanks for the explanation. That clarifies it for me.